File: cpp/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> hash;

        for (int i = 0; i < n; i++) {
            int need = target - nums[i];

            if (hash.count(need)) {
                return {hash[need], i};
            }
            hash[nums[i]] = i;
        }
        return {};
    }
};

File: kotlin/1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {

        if (nums.isEmpty()) {
            return intArrayOf()
        }

        val map = HashMap<Int, Int>()

        for (i in nums.indices) {

            val current = nums[i]

            map[current] = i

            val complement = target - current

            if (map.containsKey(complement)) {

                return intArrayOf(
                    map[complement]!!,
                    i
                )
            }
        }

        return intArrayOf(-1, -1)
    }
}